### PR TITLE
ci: use prerelease logic

### PR DESCRIPTION
**Pull request by ReenigneArcher**
## Description

- Use new pre-release logic.
- Update website to dynamically determine latest version, and latest pre-release version. Only show "beta" download button if it's newer than the latest regular release.

Todo:
- [x] Update version checking inside of Sunshine UI
- [x] Update versioning in cmake
- [x] Move winget to release event workflow
- [x] Move publish brew to release event workflow (this will also require a change to homebrew-release-action)
- [x] Update crowdin to point to master (after merge and branch renames)
- [x] Update branch protection rules

Testing version check UI:

Faked a build at v0.23.1 (current latest):
image

Faked a build at v0.23.0 (outdated):
image

Cannot really test when a pre-release is available, without creating one or two. Should work though as it's the same logic. Might have to fix forward if anything comes up.

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency update (updates to dependencies)
- [x] Documentation update (changes to documentation)
- [x] Repository update (changes to repository files, e.g. `.github/...`)

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated the in code docstring/documentation-blocks for new or existing methods/components

## Branch Updates
LizardByte requires that branches be up-to-date before merging. This means that after any PR is merged, this branch
must be updated before it can be merged. You must also
[Allow edits from maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I want maintainers to keep my branch updated

### CODE REVIEW

File: .github/workflows/update-homebrew-release.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+on:
+  release:
+    types: [created, edited]
+
+concurrency:
+  group: "${{ github.workflow }}-${{ github.event.release.tag_name }}"
+  cancel-in-progress: true
+
+jobs:
+  update-wingete-release:
```
Comment (Hazer) [inline .github/workflows/update-homebrew-release.yml:19]: This should be `update-homebrew-release`?

File: .github/workflows/update-homebrew-release.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    if: >-
+      github.repository_owner == 'LizardByte' &&
+      !github.event.release.draft && !github.event.release.prerelease
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check if Homebrew repo
+        id: isWingetRepo
```
Comment (Hazer) [inline .github/workflows/update-homebrew-release.yml:26]: Is this right?

File: .github/workflows/update-homebrew-release.yml (hunk 1 above)
Comment (Hazer) [reply .github/workflows/update-homebrew-release.yml:19]:
```suggestion
  update-homebrew-release:
```

File: .github/workflows/update-homebrew-release.yml (hunk 2 above)
Comment (Hazer) [reply .github/workflows/update-homebrew-release.yml:26]:
```suggestion
        id: isHomebrewRepo
```

**Review by Hazer — CHANGES_REQUESTED** [review]

File: src_assets/common/assets/web/index.html [hunk 3]
```diff
@@ -110,49 +107,41 @@ <h3>{{githubVersion.name}}</h3>
...
       buildVersionIsDirty() {
         return this.version?.split(".").length === 5 &&
           this.version.indexOf("dirty") !== -1
       },
-      buildVersionIsNightly() {
-        return this.version?.split(".").length === 4
-      },
-      buildVersionIsStable() {
-        return this.version?.split(".").length === 3
-      },
```
Comment (ReenigneArcher) [inline src_assets/common/assets/web/index.html:155]: There will no longer be a reliable way to determine if the user is using a pre-release or stable, so we probably need to add a config option to allow ignoring notifications for pre-releases.

Comment (Hazer) [reply src_assets/common/assets/web/index.html:155]: Can't we add a new flag to the build that just compiles "stable", "beta" or "dirty"? Given which pipeline, or lack thereof it ran?

File: src_assets/common/assets/web/configs/tabs/General.vue [hunk 4]
```diff
@@ -127,6 +127,16 @@ function removeCmd(index) {
...
     </div>
+
+    <!-- Notify Pre-Releases -->
+    <div class="mb-3">
+        <label for="notify_pre_releases" class="form-label">{{ $t('config.notify_pre_releases') }}</label>
+        <select id="notify_pre_releases" class="form-select" v-model="config.notify_pre_releases">
+            <option value="disabled">{{ $t('_common.disabled') }}</option>
+            <option value="enabled">{{ $t('_common.enabled') }}</option>
+        </select>
+        <div class="form-text">{{ $t('config.notify_pre_releases_desc') }}</div>
```
Comment (Hazer) [inline src_assets/common/assets/web/configs/tabs/General.vue:138]: How about, stable, beta and none? This way if someone on beta, want to start receiving the next stable whenever it launches, they can too

**Review by Hazer — CHANGES_REQUESTED** [review]

File: src_assets/common/assets/web/configs/tabs/General.vue (hunk 4 above)
Comment (Hazer) [reply src_assets/common/assets/web/configs/tabs/General.vue:138]: As you clarified, we aways shows stable updates, so there's no need to change channel.

**Pull request merged**